### PR TITLE
[ONNX] Support aten:normal op.

### PR DESCRIPTION
**Pull request by fatcat-z**
1. Add a new symbolic function broadcast_tensors() to support exporting torch.broadcast_tensors() function. This is required by exporting torch.distribution.normal() function.
2. Add a new symbolic function normal() to support exporting torch.distribution.normal() function.
3. Add relative tests as well.

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -9072,6 +9072,49 @@ def forward(self, x):
         x = torch.ones(2, 0, 3)
         self.run_test(M(), (x,))
 
+    def test_broat_cast_tensors(self):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:9075]: nit: type. "broadcast"

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -9072,6 +9072,49 @@ def forward(self, x):
...
+        class M(torch.nn.Module):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:9076]: Question: How does broadcast work for scalar tensors, or tensors with a '0' dimension? Can we add a test?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -9072,6 +9072,49 @@ def forward(self, x):
...
+            def forward(self, x, y):
+                return torch.distributions.Normal(x, y).sample().size(0), x, y
+
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([[1.0], [2.0]])))
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([1.0])))
+
+        self.run_test(M(), (torch.tensor([[[0.0], [10.0]], [[2.0], [8.0]], [[2.0], [8.0]]]), torch.tensor([[1.0], [3.0]])))
+
+    @disableScriptTest()
+    def test_dist_uniform(self):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:9108]: Would it be possible to add a statistical test, and verify the distributions? Maybe have a loop to generate 100 samples and compare mean and variance of samples with expected mean and variance?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:9075]: Good catch.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 2 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:9076]: For scalar tensor, it is just a tensor with 1 dimension so it works as expected.
For tensor with 0 dimension, it is not supported by broadcast yet, so not necessary to add test.

File: torch/onnx/symbolic_opset11.py [hunk 4]
```diff
@@ -983,3 +985,28 @@ def repeat_interleave(g, self, repeats, dim=None, output_size=None):
     # along the dimension provided, some post-processing is required
     loop_out = g.op("Transpose", loop_out, perm_i=perm_i)
     return reshape(g, loop_out, g.op("Constant", value_t=torch.LongTensor(output_sizes)))
+
+def broadcast_tensors(g, self):
+    all_tensors = sym_help._unpack_list(self)
```
Comment (shubhambhokare1) [inline torch/onnx/symbolic_opset11.py:990]: Maybe could add a check here along the lines
if sym_help._is_packed_list(self)

Comment (neginraoof) [reply torch/onnx/symbolic_opset11.py:990]: I agree. Do we expect the 'self' argument to be a packed_list always?

Comment (fatcat-z) [reply torch/onnx/symbolic_opset11.py:1001]: Yes, we do.

Comment (neginraoof) [reply torch/onnx/symbolic_opset11.py:1001]: Can we add assert?

**Comment by neginraoof** [comment]
Please rebase the PR

File: torch/onnx/symbolic_opset11.py (hunk 4 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_opset11.py:1001]: Since we expect the 'self' argument to be a packed_list always, we don't need to add an extra assert here. Such check has been done in sym_help._unpack_list() method already.

Per my understanding, the usage of adding an assert here is to handle the case that self argument is not a packed_list. But this is not what we expected so we should fail in such case instead of creating a new list which contains only 1 element(self argument).

Comment (neginraoof) [reply torch/onnx/symbolic_opset11.py:1001]: So assert is exactly for when we expect a behavior.
If you expect it to be packed_list, you put an assert, right?
This is not "handle" the case if input is not packed_list. It is used simply to assert the expectation, and guard for any (future) bugs or even changes (possibly in the upstream).
It just makes debugging easier. In any case, if it's already done in sym_help._unpack_list(), it's okay.

File: docs/source/onnx.rst [hunk 5]
```diff
@@ -435,6 +435,101 @@ The example below shows how you can access ``requires_grad`` via the ``Node`` ob
     from torch.onnx import register_custom_op_symbolic
     register_custom_op_symbolic("::prim_PythonOp", symbolic_pythonop, 1)
 
+Autograd Function
```
Comment (neginraoof) [inline docs/source/onnx.rst:438]: Please rebase. These file changes are unrelated.
